include/text_view_detail/otext_iterator.hpp

```diff
@@ -35,7 +32,36 @@ class otext_cursor
     using state_transition_type = typename ET::state_transition_type;
 
 public:
-    using mixin = otext_iterator_mixin<ET, CUIT>;
+    using difference_type = ranges::difference_type_t<iterator_type>;
+
+    class mixin
```

Comment:
Normally I'm not a big fan of member classes.  But in this case I agree this is a definite improvement.  I updated <tt>itext_cursor</tt> similarly in commit fff77e38eab3e91ce53ee53e4aa2f61faeb12f13.